### ISSUE TITLE
make /user/* return bookmark data

### ISSUE THREAD

**Issue by davidascher** [participedia/api#75]
too sql fancy for me to do in a single query.

**Comment by davidascher** [comment]
The current query also doesn't return enough data for authored bits.  We need, for each authored case, method, and org: `id`, `lead_image`, `title`, `type`.

Currently it gives:

```
{
    OK: true,
    data: {
        name: "Jeanne Ross",
        id: 123,
        type: "user",
        cases: [
            {
                f1: 38,
                f2: "Far North Queensland Citizen's Jury"
            }
        ],
        methods: null,
        organizations: null
    }
}
```

(ideally let's replace `f1/f2` with real key names)

**Comment by davidascher** [comment]
Got as far as this:

```
SELECT
	users.name,
	users.id,
    'user' as type,
	to_json(cases_authored) cases,
	to_json(methods_authored) methods,
	to_json(organizations_authored) organizations,
	to_json(bookmarked_cases) bookmarked_cases
FROM
	users LEFT JOIN
	(
	    SELECT DISTINCT
	         array_agg(ROW(authors.case_id, texts.title)) cases_authored, authors.user_id
	    FROM
	        case__localized_texts texts,
	        case__authors authors
	    WHERE
	        texts.language = ${language} AND
	        texts.case_id = authors.case_id
	    GROUP BY
	    	authors.user_id
        HAVING
            authors.user_id = ${userId}
	) AS case_authors ON case_authors.user_id = users.id LEFT JOIN
	(
	    SELECT DISTINCT
	         array_agg(ROW(authors.method_id, texts.title)) methods_authored, authors.user_id
	    FROM
	        method__localized_texts texts,
	        method__authors authors
	    WHERE
	        texts.language = ${language} AND
	        texts.method_id = authors.method_id
	    GROUP BY
	    	authors.user_id
        HAVING
            authors.user_id = ${userId}
	) AS method_authors ON method_authors.user_id = users.id LEFT JOIN
	(
	    SELECT DISTINCT
	         array_agg(ROW(authors.organization_id, texts.title)) organizations_authored, authors.user_id
	    FROM
	        organization__localized_texts texts,
	        organization__authors authors
	    WHERE
	        texts.language = ${language} AND
	        texts.organization_id = authors.organization_id
	    GROUP BY
	    	authors.user_id
        HAVING
            authors.user_id = ${userId}
	) AS org_authors  ON org_authors.user_id = users.id LEFT JOIN
	(
		SELECT DISTINCT
			array_agg(
				ROW(
					bookmarks.thingid, 
					bookmarks.userid,
					texts.title,
					cases.lead_image
				)
			), bookmarks.thingid, bookmarks.userid userid
		FROM 
			users,
			bookmarks,
			case__localized_texts texts,
			cases
		WHERE
			bookmarks.bookmarktype='case' AND
			bookmarks.userid=users.id AND
			texts.case_id = bookmarks.thingid AND
			cases.id = bookmarks.thingid
		GROUP BY
			bookmarks.userid, bookmarks.thingid
		HAVING	bookmarks.userid = ${userId}
	) AS bookmarked_cases 
	ON bookmarked_cases.userid = users.id
WHERE
	users.id = ${userId}
;
```

but it's returning a row per bookmark, instead of all bookmarks in an array.

**Comment by dethe** [comment]
Fixed in PR #90
